### DISCUSSION TITLE
How to deal with domain's heterogeneity ?

### DISCUSSION THREAD

**Discussion by LPSavoie** [zwicker-group/py-pde#396] [Q&A]
First of all, thank you so much for sharing py-pde and congratulations for such a beautiful code.
It is a shame though that this code doesn't seem capable of dealing with domain's heterogeneity ?  

Problem statement:
Say we apply the diffusion equation div.(k.grad(T)=0 to  a square domain of dimension 1 by 1
Boundary conditions are:
left hand side - Dirichlet / value= 1
right hand side - Dirichlet / value = 0
top and bottom sides - Neuman / derivative = 0

the domain is split in two:
0<= x <=0.5, k = 10
0.5< x <= 1, k = 1

Would py-pde allow the modeling of this simple example?

**Comment by david-zwicker** [comment]
This is certainly possible! In fact, there is [an example, which solves a very similar problem](https://py-pde.readthedocs.io/en/latest/examples_gallery/pde_heterogeneous_diffusion.html#sphx-glr-examples-gallery-pde-heterogeneous-diffusion-py).

**Reply by LPSavoie** [reply]
Thank you for the swift reply ! 
Indeed the example provides some heterogeneity in the domain, however the (1+tanh(x)) does not represent a strong / stiff discontinuity.
What about dealing with two very distinct domain properties?

**Reply by david-zwicker** [reply]
You could either make the hyperbolic tangent steeper, or you could also use a scalar field as the diffusivity via the `consts` Argument of the `PDE` class; see [the documentation](https://py-pde.readthedocs.io/en/latest/packages/pde.pdes.pde.html#pde.pdes.pde.PDE).
